### ISSUE TITLE
Should ignored-unit-patterns include parameter destructuring

### ISSUE THREAD

**Issue by cloneable** [rust-lang/rust-clippy#11403]
### Summary

error: matching over `()` is more explicit
 --> src/lib.rs:8:12
  |
8 | pub fn moo(_: ()) {}
  |            ^ help: use `()` instead of `_`: `()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ignored_unit_patterns
  = note: `-D clippy::ignored-unit-patterns` implied by `-D clippy::pedantic`

### Lint Name

ignored_unit_patterns

### Reproducer

I tried this code:

```rust
pub fn moo(_: ()) {}
```

I saw this happen:

```
error: matching over `()` is more explicit
 --> src/lib.rs:1:12
  |
8 | pub fn moo(_: ()) {}
  |            ^ help: use `()` instead of `_`: `()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ignored_unit_patterns
  = note: `-D clippy::ignored-unit-patterns` implied by `-D clippy::pedantic`
```

I expected to see this happen:

It's unclear from #11242 and the test cases if this check was intended to include parameter destructuring.

### Version

```text
rustc 1.73.0-beta.1 (680cdf816 2023-08-21)
binary: rustc
commit-hash: 680cdf8168a906b4ea80af673c64e4a16f77be57
commit-date: 2023-08-21
host: aarch64-apple-darwin
release: 1.73.0-beta.1
LLVM version: 17.0.0
```

**Comment by Alexendoo** [comment]
Yeah I think this would be good to ignore

**Comment by Centri3** [comment]
This also extends to `let` statements, where type annotations are present: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=edc8b6319b4a6ad1374ee18c17125b2a

**Comment by cloneable** [comment]
Would it be best to move this check to clippy::nursery (before next beta) until things are sorted? I do believe it should be part of clippy::pedantic in the future, at least personally I'd like to see it there, but it seems this check needs more work.

**Comment by cloneable** [comment]
Sorry, I noticed that beta.5 went out without this fix. Was this the last beta before the release? Does this mean this lint gets included in pedantic in the unfixed state in 1.73?
